src/main/java/de/tum/in/www1/artemis/service/user/UserService.java

```diff
@@ -486,6 +497,26 @@ protected void deleteUser(User user) {
         userRepository.flush();
     }
 
+    /**
+     * Sets the properties of the user to random or dummy values, making it impossible to identify the user.
+     *
+     * @param user the user that should be anonymized
+     */
+    protected void anonymizeUser(User user) {
+        user.setFirstName(Constants.USER_FIRST_NAME_AFTER_SOFT_DELETE);
+        user.setLastName(Constants.USER_LAST_NAME_AFTER_SOFT_DELETE);
+        user.setLogin(RandomUtil.generateRandomAlphanumericString());
+        user.setPassword(RandomUtil.generatePassword());
+        user.setEmail(Constants.USER_EMAIL_AFTER_SOFT_DELETE);
+        user.setRegistrationNumber("");
+        user.setImageUrl("");
```

Comment:
```suggestion
        user.setRegistrationNumber(null);
        user.setImageUrl(null);
```
Nullable information should be `null` where it is no longer needed.